pkcs11/_pkcs11.pyx

```diff
@@ -1161,15 +1168,44 @@ cdef class lib:
     cdef public tuple cryptoki_version
     cdef public tuple library_version
 
-    def __cinit__(self):
-        assertRV(C_Initialize(NULL))
+    def _load_pkcs11_lib(self, so):
+        """Load a PKCS#11 library, and extract function calls.
+
+        This method will dynamically load a PKCS11 library, and attempt to
+        resolve the symbol 'C_GetFunctionList()'. Once found, the entry point
+        is called to populate an internal table of function pointers.
+
+        This is a private method, and must never be called directly.
+        Called when a new lib class is instantiated.
+
+        :param so: the path to a valid PKCS#11 library
+        :type so: str
+        :raises: RuntimeError or PKCS11Error
+        :rtype: None
+        """
+
+        # to keep a pointer to the C_GetFunctionList address returned by dlsym()
+        cdef C_GetFunctionList_p C_GetFunctionList
+
+        handle = dlfcn.dlopen(so.encode('utf-8'), dlfcn.RTLD_LAZY | dlfcn.RTLD_LOCAL)
+        if handle == NULL:
+            raise RuntimeError(dlfcn.dlerror())
+
+        C_GetFunctionList = <C_GetFunctionList_p>dlfcn.dlsym(handle, 'C_GetFunctionList')
+        if C_GetFunctionList==NULL:
+            raise RuntimeError("{} is not a PKCS#11 library: {}".format(so,dlfcn.dlerror()))
```

Comment:
...yes you are 😅 It's OK, "readability counts", doesn't it? I haven't found a linter to run against Cython files yet.
for the cast , you really meant "after" rather than "before" ? I made this change, let me know if that suits your requirement.
```python
C_GetFunctionList = <C_GetFunctionList_ptr> dlfcn.dlsym(handle, 'C_GetFunctionList')
```